### ISSUE TITLE
Docker CPU image cntk:2.2-cpu-python3.5 BROKEN

### ISSUE THREAD

**Issue by bgie** [Microsoft/CNTK#2632]
Using docker for windows on win10
No GPU, using docker image microsoft/cntk:2.2-cpu-python3.5
started it with 
`  docker run -p 8888:8888 -it microsoft/cntk:2.2-cpu-python3.5 /bin/bash`
Then in my bash I get the message "CNTK is activated."
Launching jupyter with:
`  jupyter-notebook --no-browser --port=8888 --ip=0.0.0.0 --notebook-dir=/cntk/Tutorials --allow-root`
First example I open:
`  CNTK_101_LogisticRegression`
When I run the steps, at step 6 I get:
```
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
~/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py in swig_import_helper()
     17         try:
---> 18             return importlib.import_module(mname)
     19         except ImportError:

~/anaconda3/envs/cntk-py35/lib/python3.5/importlib/__init__.py in import_module(name, package)
    125             level += 1
--> 126     return _bootstrap._gcd_import(name[level:], package, level)
    127 

~/anaconda3/envs/cntk-py35/lib/python3.5/importlib/_bootstrap.py in _gcd_import(name, package, level)

~/anaconda3/envs/cntk-py35/lib/python3.5/importlib/_bootstrap.py in _find_and_load(name, import_)

~/anaconda3/envs/cntk-py35/lib/python3.5/importlib/_bootstrap.py in _find_and_load_unlocked(name, import_)

ImportError: No module named 'cntk._cntk_py'

During handling of the above exception, another exception occurred:

ImportError                               Traceback (most recent call last)
<ipython-input-6-32eb5c67939c> in <module>()
     12     from urllib import urlopen
     13 
---> 14 import cntk as C

~/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/__init__.py in <module>()
      8 import numpy as np
      9 
---> 10 from . import cntk_py
     11 
     12 #

~/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py in <module>()
     19         except ImportError:
     20             return importlib.import_module('_cntk_py')
---> 21     _cntk_py = swig_import_helper()
     22     del swig_import_helper
     23 elif _swig_python_version_info >= (2, 6, 0):

~/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py in swig_import_helper()
     18             return importlib.import_module(mname)
     19         except ImportError:
---> 20             return importlib.import_module('_cntk_py')
     21     _cntk_py = swig_import_helper()
     22     del swig_import_helper

~/anaconda3/envs/cntk-py35/lib/python3.5/importlib/__init__.py in import_module(name, package)
    124                 break
    125             level += 1
--> 126     return _bootstrap._gcd_import(name[level:], package, level)
    127 
    128 

ImportError: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
```

All following steps fail too.

**Comment by bgie** [comment]
Outside of jupyter notebook same problem, the entire cntk install in the container is broken.
Opened a bash into the container, freshly downloaded... Made sure I did:
  `source` /cntk/activate-cntk

From bash, running:
`python` -c "import cntk; print(cntk.__version__)"
Gives:
```
Traceback (most recent call last):
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py", line 18, in swig_import_helper
    return importlib.import_module(mname)
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 956, in _find_and_load_unlocked
ImportError: No module named 'cntk._cntk_py'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/__init__.py", line 10, in <module>
    from . import cntk_py
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py", line 21, in <module>
    _cntk_py = swig_import_helper()
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/site-packages/cntk/cntk_py.py", line 20, in swig_import_helper
    return importlib.import_module('_cntk_py')
  File "/root/anaconda3/envs/cntk-py35/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: libnvidia-ml.so.1: cannot open shared object file: No such file or directory
```

**Comment by hellosaumil** [comment]
I can provide you a temporary fix: #1917 

```
mv /cntk/cntk/lib /cntk/cntk/lib-old
pip install --upgrade https://cntk.ai/PythonWheel/CPU-Only/cntk-2.2-cp35-cp35mu-linux_x86_64.whl
```

One drawback that I'm facing is as soon as I _restart_ the docker container, I have to repeat this process again...until this issue is solved try this fix.

**Comment by KeDengMS** [comment]
Thanks for the workaround. You can run 'docker ps -a' to find a stopped docker container, and run 'docker start <container name>' to recover it, instead of 'docker run' to start a new one. That way no repeated fix needed.

**Comment by hellosaumil** [comment]
@KeDengMS I'm actually using Kitematic to Start or Stop Container...but I'll check this out and respond whether it retains the state.

**Comment by hellosaumil** [comment]
@KeDengMS Docker Start/Stop works instead of running/restarting from KinteMatic Docker App on macOS. Thanks! 👍

**Comment by hellosaumil** [comment]
@KeDengMS There is one more issue after this fix. 

I was able to run cant perfectly in Python Interpreter or even in Jupyter Notebook.

But when I tried to create Brainscript using command `cntk configFile=01_OneHidden_ndl_deprecated.cntk` from [CNTK usage overview](https://docs.microsoft.com/en-us/cognitive-toolkit/CNTK-usage-overview)...

OR 

Just write `cntk` in bash of the docker container.

I ran into the same error as before:
`cntk: error while loading shared libraries: libnvidia-ml.so.1: cannot open shared object file: No such file or directory`

**Comment by KeDengMS** [comment]
Can you try the latest 2.3 release? This issue is fixed in 2.3

**Comment by bgie** [comment]
I've tested the new 2.3 image, can confirm issue is fixed.

Did this:
`docker run -p 8888:8888 -it microsoft/cntk:2.3-cpu-python3.5 /bin/bash`

Then:
`python -c "import cntk; print(cntk.__version__)"`
which correctly gives:
`2.3`

The tutorials in the jupyter notebook work too.

Thans!
